### ISSUE TITLE
特定位置、角度下，mrcd:ray_tick 递归次数超过500 | mrcd:ray_tick function exceeds the recursion limit of 500 at a specific position and angle

### ISSUE THREAD

**Issue by Youmiel** [K-bai/Wonderful-Raycast-system-Near-My-Door#7]
- Minecraft 版本：1.18.2
- MRCD 版本：2.6.3

测试代码：
```
execute at @s anchored eyes run summon marker ^ ^ ^32 {Tags:["mrcd_bullet","init"]}

execute as @e[tag=init, limit=1] store result score @s mrcd_x0 run data get entity @s Pos[0] 1000
execute as @e[tag=init, limit=1] store result score @s mrcd_y0 run data get entity @s Pos[1] 1000
execute as @e[tag=init, limit=1] store result score @s mrcd_z0 run data get entity @s Pos[2] 1000

execute at @s anchored eyes run tp @e[tag=init, limit=1] ^ ^ ^ ~ ~
execute as @e[tag=init, limit=1] store result score #temp_x mrcd_system run data get entity @s Pos[0] 1000
execute as @e[tag=init, limit=1] store result score #temp_y mrcd_system run data get entity @s Pos[1] 1000
execute as @e[tag=init, limit=1] store result score #temp_z mrcd_system run data get entity @s Pos[2] 1000

scoreboard players operation @e[tag=init, limit=1] mrcd_x0 -= #temp_x mrcd_system
scoreboard players operation @e[tag=init, limit=1] mrcd_y0 -= #temp_y mrcd_system
scoreboard players operation @e[tag=init, limit=1] mrcd_z0 -= #temp_z mrcd_system

execute as @e[tag=init, limit=1] at @s run function mrcd:ray_tick

tag @e[tag=mrcd_target_entity] remove mrcd_target_entity
kill @e[tag=init]
```

已知稳定复现点位：
- `execute in minecraft:overworld run tp @s 17.13 -59.72 22.3 45.99 -0.0055` 至
- `execute in minecraft:overworld run tp @s 17.13 -59.72 22.3 45.99 -0.049`

**Comment by mattcarter11** [comment]
Hi @Youmiel,

this is a problem I was aware the datapack had. But I was able to fix it in 2.7 (plus i optimized a lot the pack)

![image](https://github.com/K-bai/Minecraft-Ray-Collision-Detector/assets/38189440/a23f3fb2-9e3d-4a56-9909-d9974f338dd5)

Just update to the 2.7.1 version to fix the problem. Tell me if the the problem persists or not. To ensure it's all ok.

Thanks

**Comment by Youmiel** [comment]
就我测试来说，目前没有发现问题。
